ash/src/extensions/khr/get_surface_capabilities2.rs

```diff
@@ -0,0 +1,55 @@
+use crate::prelude::*;
+use crate::vk;
+use crate::{Entry, Instance};
+use std::ffi::CStr;
+use std::mem;
+
+#[derive(Clone)]
+pub struct GetSurfaceCapabilities2 {
+    fp: vk::KhrGetSurfaceCapabilities2Fn,
+}
+
+impl GetSurfaceCapabilities2 {
+    pub fn new(entry: &Entry, instance: &Instance) -> Self {
+        let fp = vk::KhrGetSurfaceCapabilities2Fn::load(|name| unsafe {
+            mem::transmute(entry.get_instance_proc_addr(instance.handle(), name.as_ptr()))
+        });
+        Self { fp }
+    }
+
+    #[doc = "https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetPhysicalDeviceSurfaceCapabilities2KHR.html"]
+    pub unsafe fn get_physical_device_surface_capabilities2(
+        &self,
+        physical_device: vk::PhysicalDevice,
+        surface_info: &vk::PhysicalDeviceSurfaceInfo2KHR,
+    ) -> VkResult<vk::SurfaceCapabilities2KHR> {
+        let mut surface_capabilities = Default::default();
+        self.fp
+            .get_physical_device_surface_capabilities2_khr(
+                physical_device,
+                surface_info,
+                &mut surface_capabilities,
+            )
+            .result_with_success(surface_capabilities)
+    }
+
+    #[doc = "https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetPhysicalDeviceSurfaceFormats2KHR.html"]
+    pub unsafe fn get_physical_device_surface_formats2(
+        &self,
+        physical_device: vk::PhysicalDevice,
+        surface_info: &vk::PhysicalDeviceSurfaceInfo2KHR,
+    ) -> VkResult<Vec<vk::SurfaceFormat2KHR>> {
+        read_into_uninitialized_vector(|count, data| {
```

Comment:
`vk::SurfaceFormat2KHR` has an `s_type` and `p_next` (unlike `vk::SurfaceFormatKHR`) hence needs to be initialized with `read_into_defaulted_vector` here. Perhaps this `p_next` should be settable externally just like other call-sites that have a separate `_len()` function (see for example `fn get_device_image_sparse_memory_requirements` in `maintenance4`).

Reply:
Good points that I overlooked. Have made the suggested changes, let me know if I'm handling the errors correctly.